src/plugins/intel_cpu/src/nodes/fullyconnected.cpp

```diff
@@ -668,6 +688,73 @@ InferenceEngine::Precision FullyConnected::getRuntimePrecision() const {
     return getMaxPrecision(inputPrecisions);
 }
 
+bool FullyConnected::canUseSparseWeights() {
+    if (useSparseWeights == false) {
+        return useSparseWeights;
+    }
+
+    if (!impl::cpu::x64::mayiuse(impl::cpu::x64::avx512_core_amx))
+        return false;
+
+    if (isDynamicNode()) {
+        return false;
+    }
+
+    auto inputDataType = DnnlExtensionUtils::IEPrecisionToDataType(getOriginalInputPrecisionAtPort(DATA_ID));
+    auto weightsDataType = DnnlExtensionUtils::IEPrecisionToDataType(getOriginalInputPrecisionAtPort(WEIGHTS_ID));
+    if (!one_of(inputDataType , memory::data_type::u8, memory::data_type::s8) || weightsDataType != memory::data_type::s8) {
+        return false;
+    }
+
+    // [todo] delete
+    printSparseRates = false; // useSparseWeights;
+
+    // calculate sparse rate
+    const auto constNode = std::dynamic_pointer_cast<Input>(getParentEdgeAt(WEIGHTS_ID)->getParent());
+    if (!constNode) {
+        IE_THROW() << "Cannot cast " << WEIGHTS_ID << " input to Input node for " << getName() << ".";
+    }
+    auto blb = constNode->getMemoryPtr();
+    if (blb == nullptr)
+        IE_THROW() << "Cannot get const blob for node " << getName() << ".";
+
+    auto weightsData = reinterpret_cast<const int8_t*>(blb->GetPtr());
+    auto elementsCount = blb->GetDescWithType<BlockedMemoryDesc>()->getPaddedElementsCount();
+    size_t zerosCounts = 0;
+    for (int i = 0; i < elementsCount; i++) {
+        if (weightsData[i] == 0) {
+            zerosCounts++;
+        }
+    }
+    nnzCount = elementsCount - zerosCounts;
+
+    if (printSparseRates) {
+        std::cout << getName() << ", weightsData.size() = " << elementsCount << ", zerosCounts = "
+            << zerosCounts << ", nnzCount = " << nnzCount << std::endl;
+    }
+
+    weiSparseRate = static_cast<float>(zerosCounts) / static_cast<float>(elementsCount);
+
+    // minSparseRate == 1 means that user did not specify min sparse rate and minSparseRate will be set automatically
+    // todo: [av] maybe change to -1?
+    if (minSparseRate == 1.f) {
```

Comment:
Now the behavior should be adjusted. `minSparseRate == 1.0f` - default value that means sparsity is switched off. In that case we should keep load time of current path: canUseSparseWeights should return false in very beginning,

Reply:
changed, thanks